turbinia/workers/file_system_timeline.py

```
@@ -67,25 +68,42 @@ def run(self, evidence, result):
-    except dfvfs_errors.ScannerError as exception:
-      result.log('Error generating bodyfile {0!s}'.format(exception))
-      status = 'Unable to generate bodyfile using provided evidence data.'
+      file_entries = None
+      with open(bodyfile_output, 'w', encoding='utf-8') as file_object:
+        try:
+          file_entries = enumerate(
+              entry_lister.ListFileEntries(base_path_specs))
+          while file_entries:
+            try:
+              _, (file_entry, path_segments) = next(file_entries)
+              bodyfile_entries = entry_lister.GetBodyfileEntries(
+                  file_entry, path_segments)
+              for bodyfile_entry in bodyfile_entries:
+                file_object.write(bodyfile_entry)
+                file_object.write('\n')
+                number_of_entries += 1
+            except StopIteration:
+              break
+        except (dfvfs_errors.AccessError, dfvfs_errors.BackEndError,
+                dfvfs_errors.MountPointError,
+                dfvfs_errors.PathSpecError) as exception:
+          status = 'Unable to process file entry: {0!s}'.format(exception)
+          result.log(status)
+
+      if number_of_entries > 0:
+        output_evidence.number_of_entries = number_of_entries
+        result.add_evidence(output_evidence, evidence.config)
+        status = 'Generated file system timeline containing [{0:d}] entries'.format(
+            number_of_entries)
+        result.close(self, success=True, status=status)
+      else:
+        status = 'Unable to process any file entries: {0!s}'.format(exception)
+        result.close(self, success=False, status=status)
+
+    except IOError as exception:
+      status = 'Unable to create bodyfile local output file: {0!s}'.format(
+          exception)
       result.close(self, success=False, status=status)
-      raise TurbiniaException(
-          'Could not process volume: {0!s}'.format(exception))
+      raise TurbiniaException(status) from exception
```

I know this was there prior to this change, but I don't think we actually need to raise an exception here, we can just return the result since you've already closed out the result and set the correct attributes indicating that it failed.

---

Done.